### PR TITLE
perf(505): Track the number of rows in each table

### LINKED ISSUE #505: Add row count metric for each table
From @kurtismullins:

There are currently metrics around the time it takes to do various operations to tables. To shine a little context on these operations, lets expose a metric which simply states how many rows a given table has.

If this already exists, feel free to just let me know and close this issue. I couldn't find it but may have missed it. An example metric which would be more useful if we knew the row count is: `spacetime_rdb_iter_time_count`.

### PR DESCRIPTION
**Pull request by joshua-spacetime**
Closes #505.

# Description of Changes

# API and ABI

 - [ ] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

# Expected complexity level and risk

*How complicated do you think these changes are? Grade on a scale from 1 to 5,
where 1 is a trivial change, and 5 is a deep-reaching and complex change.*

*This complexity rating applies not only to the complexity apparent in the diff,
but also to its interactions with existing and future code.*

*If you answered more than a 2, explain what is complex about the PR,
and what other components it interacts with in potentially concerning ways.*

### CODE REVIEW

File: crates/core/src/db/commit_log.rs [hunk 1]
```diff
@@ -111,33 +111,19 @@ impl CommitLog {
         let mut unwritten_commit = self.unwritten_commit.lock().unwrap();
         let mut writes = Vec::with_capacity(tx_data.records.len());
 
-        let rows_inserted = &DB_METRICS.rdb_num_rows_inserted;
-        let rows_deleted = &DB_METRICS.rdb_num_rows_deleted;
+        let mut inserts = HashMap::new();
```
[the remark this answers is not in the document]
Comment (cloutiertyler) [reply crates/core/src/db/commit_log.rs:114]: I worry a little bit about the perf implications for generating a copy here as well, but it's probably fine?

**Review by cloutiertyler — COMMENTED** [review]
Just one nit.

File: crates/core/src/db/datastore/locking_tx_datastore/mod.rs [hunk 2]
```diff
@@ -1569,10 +1597,34 @@ impl Locking {
 
         // Create the system tables and insert information about themselves into
         // st_table, st_columns, st_indexes, and st_sequences.
+        DB_METRICS
+            .rdb_num_table_rows
+            .with_label_values(&database_address, &st_table_schema().table_id.into())
```
Comment (mamcx) [inline crates/core/src/db/datastore/locking_tx_datastore/mod.rs:1602]: The `table_id` of these objects could be queried once?

**Review by mamcx — APPROVED** [review]
LGTM

**Pull request merged**